feature-settings-impl/src/main/java/io/novafoundation/nova/feature_settings_impl/domain/CustomNodeInteractor.kt

```diff
@@ -0,0 +1,52 @@
+package io.novafoundation.nova.feature_settings_impl.domain
+
+import io.novafoundation.nova.common.validation.ValidationSystem
+import io.novafoundation.nova.feature_settings_impl.data.NodeChainIdRepositoryFactory
+import io.novafoundation.nova.feature_settings_impl.domain.validation.NetworkNodeValidationSystem
+import io.novafoundation.nova.feature_settings_impl.domain.validation.validNodeUrl
+import io.novafoundation.nova.feature_settings_impl.domain.validation.validateNodeNotAdded
+import io.novafoundation.nova.runtime.multiNetwork.ChainRegistry
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import io.novafoundation.nova.runtime.repository.ChainNodeRepository
+
+interface CustomNodeInteractor {
+
+    suspend fun getNodeDetails(chainId: String, nodeUrl: String): Result<Chain.Node>
+
+    suspend fun createNode(chainId: String, url: String, name: String)
+
+    suspend fun saveNode(chainId: String, oldUrl: String, url: String, name: String)
```

Comment:
I suggest to rename it to `updateNode`, otherwise it looks a bit confusing since we have `createNode` and `saveNode`